### PR TITLE
update rule element for Sparkling Targe to account for other traits

### LINKED ISSUE #15304: Shield Block does not recalculate Rule Elements for the shield
When you use Shield Block it does not seem to recalculate Rule Elements for the shield that is blocking.

For example the Magus' Sparkling Targe should add 2 to Hardness to the Steel Shield in this case (should be 10, not 8), but it isn't added automatically:

image

image

### LINKED ISSUE #15305: Sparkling Targe rule elements only check for the Magical trait
The rule elements for the Magus class feature Sparkling Targe only check for the Magical trait, but not for Arcane, Divine, Occult, or Primal, so it isn't always automatically triggered, when it could. Reporting as per https://github.com/foundryvtt/pf2e/issues/15302#issuecomment-2198633172

### PR DESCRIPTION
**Pull request by ammalagonc**
Closes #15304
Closes #15305

### CODE REVIEW

File: packs/classfeatures/sparkling-targe.json [hunk 1]
```diff
@@ -44,6 +44,14 @@
...
                         "or": [
                             "magical",
+                            "arcane",
+                            "divine",
+                            "occult",
+                            "primal",
+                            "item:trait:arcane",
+                            "item:trait:divine",
+                            "item:trait:occult",
+                            "item:trait:primal",
```
Comment (TikaelSol) [inline packs/classfeatures/sparkling-targe.json:54]: I'm not big on including both of these. We should probably be moving to only the `item:trait:x` ones.

**Comment by iBoMbY** [comment]
This doesn't solve the issue https://github.com/foundryvtt/pf2e/issues/15304 though

File: packs/classfeatures/sparkling-targe.json [hunk 2]
```diff
@@ -44,6 +44,10 @@
                     {
                         "or": [
                             "magical",
+                            "item:trait:arcane",
+                            "item:trait:divine",
+                            "item:trait:occult",
+                            "item:trait:primal",
```
Comment (stwlam) [inline packs/classfeatures/sparkling-targe.json:50]:
```suggestion
                            "item:magical"
```

**Comment by iBoMbY** [comment]
Also Sparkling Targe does work against all magical effects, not just item effects. The first commit was correct imho.

**Comment by stwlam** [comment]
> Also Sparkling Targe does work against all magical effects, not just item effects. The first commit was correct imho.

"item" is in a foundry sense rather than a pathfinder one.

**Comment by iBoMbY** [comment]
I see a different problem here though, I did try it with "item:trait:magical" and it seems always be set/true if the shield itself is magical, even though the item/effect it defends against isn't magical. I guess in this case there would be no way to differentiate between both sides.

**Pull request merged**